compiler/dyno/lib/uast/Builder.cpp

```diff
@@ -280,12 +294,136 @@ void Builder::doAssignIDs(AstNode* ast, UniqueString symbolPath, int& i,
+    for (auto configPair: configs) {
+      if ((var->name().str() == configPair.first && var->visibility() != Decl::PRIVATE)
+          || configPair.first == possibleModule + var->name().str()) {
+        // found a config that was set via cmd line
+        // handle deprecations
+        if (auto attribs = var->attributes()) {
+          if (attribs->isDeprecated()) {
+            // TODO: Need proper message handling here
+            std::string msg = "'" + var->name().str() + "' was set via a compiler flag";
+            if (attribs->deprecationMessage().isEmpty()) {
+              std::cerr << "warning: " + var->name().str() + " is deprecated" << std::endl;
+            } else {
+              std::cerr << "warning: " + attribs->deprecationMessage().str() << std::endl;
+            }
+            std::cerr << "note: " + msg << std::endl;
+          }
+        }
+        if (!configMatched.first.empty() && configMatched.first != configPair.first) {
+          std::string errMsg = "config set ambiguously via '-s";
+          errMsg += configMatched.first;
+          errMsg += "' and '-s";
+          errMsg += configPair.first + "'";
+          ErrorMessage errorMessage = ErrorMessage(ast->id(), notedLocations_[ast], errMsg, ErrorMessage::ERROR);
+          addError(errorMessage);
+        }
+        configMatched = configPair;
+      }
+    }
+    return configMatched;
+  }
+
+  AstNode* Builder::checkAndUpdateConfig(AstNode* ast, std::pair<std::string,std::string> configPair) {
+    AstNode* ret = nullptr;
+    assert(ast->isVariable() && ast->toVariable()->isConfig());
+    assert(!configPair.first.empty());
+    auto var = ast->toVariable();
+    auto parser = parsing::Parser::build(context());
+    parsing::Parser *p = parser.get();
+    std::string inputText;
+    // for types, it's important for the parser to see that it's a type 
+    if (var->kind() == uast::Variable::TYPE) {
+      inputText = "type " + var->name().str() + "=" + configPair.second + ";";
+    } else {
+      inputText = "const " + var->name().str() + "=";
+      inputText += (!configPair.second.empty()) ? configPair.second : "true";
+      inputText += ";";
+    }
+    // TODO: how to handle nested module configs e.g., -sFoo.Baz.bar=10
+    auto parseResult = p->parseString("<command line>", inputText.c_str());
+    // TODO: add better error handling
+    assert(!parseResult.numErrors());
+    auto mod = parseResult.singleModule();
+    assert(mod);
+    owned<AstNode> initNode;
+    if (mod->stmt(0)->isVariable()) {
+      // steal the init expression, children_ will have nullptr in place
+      initNode = std::move(mod->children_[0]->children_.back());
+      // clean out the nullptr
+      mod->children_[0]->children_.pop_back();
+    } else {
+      assert(false && "should only be an assignment or type initializer");
+    }
+    // TODO: How to handle locations? Right now we are just using the same location as the config
```

Comment:
The concern covered by the TODO is that each of the child nodes will have the same (identical) location as the parent, rather than and updated location with updated column numbers. There isn't a clearly good answer here, and @mppf suggested leaving it for now in https://github.com/chapel-lang/chapel/pull/19643#discussion_r853386532